python/ray/air/_internal/accelerator_utils/nvidia_gpu.py

```diff
@@ -0,0 +1,78 @@
+import os
+from typing import List
+
+import torch
+
+import ray
+from ray.air._internal.accelerator_utils.device_manager import TorchDeviceManager
+
+
+class CUDATorchDeviceMananger(TorchDeviceManager):
+    """CUDA device manager"""
+
+    @staticmethod
+    def get_accelerator_name() -> str:
+        return "GPU"
+
+    @staticmethod
+    def get_device_type() -> str:
+        return "cuda"
+
+    @staticmethod
+    def get_device_module():
+        return torch.cuda
+
+    @staticmethod
+    def is_device_available() -> bool():
+        return torch.cuda.is_available()
+
+    @staticmethod
+    def get_devices() -> List[torch.device]:
+        """Gets the correct torch device list configured for this process.
+
+        Returns a list of torch CUDA devices allocated for the current worker.
+        If no GPUs are assigned, then it returns a list with a single CPU device.
+
+        Assumes that `CUDA_VISIBLE_DEVICES` is set and is a
+        superset of the `ray.get_gpu_ids()`.
+        """
+        if torch.cuda.is_available():
+            # GPU IDs are assigned by Ray after you specify "use_gpu"
+            # GPU `ray.get_gpu_ids()` may return ints or may return strings.
+            # We should always convert to strings.
+            gpu_ids = [str(id) for id in ray.get_gpu_ids()]
+
+            device_ids = []
+
+            if len(gpu_ids) > 0:
+                cuda_visible_str = os.environ.get("CUDA_VISIBLE_DEVICES", "")
+                if cuda_visible_str and cuda_visible_str != "NoDevFiles":
+                    cuda_visible_list = cuda_visible_str.split(",")
+                else:
+                    cuda_visible_list = []
+
+                # By default, there should only be one GPU ID if `use_gpu=True`.
+                # If there are multiple GPUs, return a list of devices.
+                # If using fractional GPUs, these IDs are not guaranteed
+                # to be unique across different processes.
+                for gpu_id in gpu_ids:
+                    try:
+                        device_ids.append(cuda_visible_list.index(gpu_id))
+                    except IndexError:
+                        raise RuntimeError(
+                            "CUDA_VISIBLE_DEVICES set incorrectly. "
+                            f"Got {cuda_visible_str}, expected to include {gpu_id}. "
+                            "Did you override the `CUDA_VISIBLE_DEVICES` environment"
+                            " variable? If not, please help file an issue on Github."
+                        )
+
+            else:
+                # If called on the driver or outside of Ray Train, return the
+                # 0th device.
+                device_ids.append(0)
+
+            devices = [torch.device(f"cuda:{device_id}") for device_id in device_ids]
+        else:
+            devices = [torch.device("cpu")]
```

Comment:
Same here：

> Thinking out loud. We should consider adding cpu as one of the accelerator type, and always have a device manager instead of having this fallback logic.

Reply:
Let's also have a `CPUTorchDeviceManager`.